app/controllers/classroom_controller.js

```diff
@@ -36,7 +36,11 @@ export const getClassrooms = (req, res) => {
 };
 
 export const getClassroom = (req, res) => {
-  ClassroomModel.findById(req.params.id).populate('teacher students')
+  // if req.user doesn't have a classroom with that id, return 422 unauthorized
+  if (req.user !== req.params.teacher) {
```

Comment:
so the `req` that will get passed into `getClassroom` will only have an `id` without any other information. the check that you actually want to do is to see if the `req.params.id` matches one of the classroom ids in `req.user.teacherClassrooms`